src/metabase/search/impl.clj

```diff
@@ -311,16 +311,14 @@
      (deferred-tru "Content Management or Official Collections")))
   (let [models (if (string? models) [models] models)
         engine (parse-engine search-engine)
-        ctx    (cond-> {:archived?                           (boolean archived)
+        fvalue (fn [filter-key] (search.config/filter-default engine context filter-key))
+        ctx    (cond-> {:archived?                           (boolean (or archived (fvalue :archived)))
```

Comment:
This doesn't change the behavior, but its nice to make the default more explicit.